components/Buttons/examples/ButtonsStoryboardAndProgrammatic.swift

```diff
@@ -80,17 +80,39 @@ class ButtonsSwiftAndStoryboardController: UIViewController {
 
     buttonSetup()
 
-    let floatingPlusShapeLayer = ButtonsTypicalUseSupplemental.createPlusShapeLayer(floatingButton)
-    floatingButton.layer.addSublayer(floatingPlusShapeLayer)
+    let plusIcon = UIImage.init(named: "Plus")?.withRenderingMode(.alwaysOriginal)
+    floatingButton.setImage(plusIcon, for: .normal)
     innerContainerView.addSubview(floatingButton)
 
-    let storyboardPlusShapeLayer =
-      ButtonsTypicalUseSupplemental.createPlusShapeLayer(floatingButton)
-    storyboardFloating.layer.addSublayer(storyboardPlusShapeLayer)
+    storyboardFloating.setImage(plusIcon, for: .normal)
 
     addButtonConstraints()
   }
 
+  func updateFloatingButtons(to mode: MDCFloatingButtonMode) {
+    if (floatingButton.mode != mode) {
+      floatingButton.mode = mode
+    }
+    if (storyboardFloating.mode != mode) {
+      storyboardFloating.mode = mode
+    }
+  }
+
+  override func viewWillAppear(_ animated: Bool) {
+    super.viewWillAppear(animated)
+    let horizontalSizeClass = self.traitCollection.horizontalSizeClass
+    let verticalSizeClass = self.traitCollection.verticalSizeClass
+    if (horizontalSizeClass == .regular && verticalSizeClass == .regular) {
+      self.updateFloatingButtons(to: .extended)
+      self.floatingButton.setTitle("Button", for: .normal)
+      self.storyboardFloating.setTitle("Button", for: .normal)
+    } else {
+      self.updateFloatingButtons(to: .normal)
+      self.floatingButton.setTitle(nil, for: .normal)
+      self.storyboardFloating.setTitle(nil, for: .normal)
+    }
```

Comment:
nit: due to the 3 line duplication for both conditionals and the same lines reused in the file in lines 240+ it might be worth putting the lines in a helper function with an input parameter for the size class to reduce code

Reply:
Done